components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -3145,14 +3167,8 @@ public static String getIdTokenIssuer(String tenantDomain) throws IdentityOAuth2
         FederatedAuthenticatorConfig oidcAuthenticatorConfig =
                 IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,
                         IdentityApplicationConstants.Authenticator.OIDC.NAME);
-        String idpEntityId = IdentityApplicationManagementUtil.getProperty(oidcAuthenticatorConfig.getProperties(),
+        return IdentityApplicationManagementUtil.getProperty(oidcAuthenticatorConfig.getProperties(),
                 IDP_ENTITY_ID).getValue();
-        if (StringUtils.isNotBlank(idpEntityId) && idpEntityId.equals(IdentityUtil.getProperty("OAuth.OpenIDConnect" +
-                ".IDTokenIssuerID"))) {
-            return IdentityUtil.resolveURL(idpEntityId, tenantDomain, true, false, false, false);
-        } else {
-            return idpEntityId;
-        }
```

Comment:
who handles the logic when the value in db is empty

Reply:
The idPEntityId is written during the tenant creation and can't be empty